src/conn.c

```diff
@@ -1410,6 +1411,10 @@ _doReconnect(void *arg)
 
     natsConn_Lock(nc);
 
+    // For external event loop, wait for the socket to be closed.
+    while ((nc->opts->evLoop != NULL) && !nc->el.sockClosed)
+        natsCondition_Wait(nc->el.cond, nc->mu);
```

Comment:
Should the other side to a broadcast and not just a directed signal?

Reply:
We are in the same connection object and there can be only a reconnect thread started, so signal is enough.

Reply:
ok just wanted to make sure we could not have a weird situation where multiple are in Wait().

Reply:
Not multiple, but that forced me to take a closer look to make sure that we don't start multiple time the _doReconnect thread when I realize that there is the condition of "retry on failed connect" in which case _doReconnect should not wait to be notified that the socket is closed, because the socket was never "opened". So added a test for that and the fix. Thanks!